### PR TITLE
Dynamically determine image of plugins

### PR DESCRIPTION
**Pull request by wilsonehusin**
**What this PR does / why we need it**:
- Determine images required by plugins dynamically using `e2e.test -list-images`
- Removing hard-coded list of images for e2e

**Which issue(s) this PR fixes**
- Partially #1191

**Special notes for your reviewer**:
We may not need to do `docker run --rm` every time by caching the results. However, I think that's best addressed in a separate PR as we have discussed about having externalized persistent config.

**TODO** :
- [x] add tests to `cmd/sonobuoy/app/images_dynamic_test.go`

**Release note**:
```
Sonobuoy no longer bundles list of images required by e2e / conformance. The required images will now be determined by `k8s.gcr.io/conformance` image at runtime. This also means that all operations in `sonobuoy images` and its subcommands now attempts to pull `k8s.gcr.io/conformance` and executes `docker run --rm` internally.
```

### CODE REVIEW

File: kind-config.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 kind: Cluster
-apiVersion: kind.sigs.k8s.io/v1alpha3
+apiVersion: kind.x-k8s.io/v1alpha4
 nodes:
 - role: control-plane
-  image: sonobuoy/kind-node:v1.19.0
+  image: &kind_node_image kindest/node:v1.19.3
```
Comment (vladimirvivien) [inline kind-config.yaml:5]: Why this change to `&kind_node_image kindest/node...`

File: pkg/image/dryrun_client.go [hunk 2]
```diff
@@ -18,12 +18,57 @@ package image
...
 import (
 	"github.com/sirupsen/logrus"
+	"strings"
 )
 
 // DryRunClient is an implementation of Client that logs the image operations that would
 // be performed rather than performing them.
 type DryRunClient struct{}
 
+const v1_19_images = `docker.io/library/httpd:2.4.38-alpine
```
Comment (vladimirvivien) [inline pkg/image/dryrun_client.go:28]: I did not expect to see this hard coded list of images... seeing this PR is to solve that.  Is this const for a test ?

File: pkg/image/docker_client.go [hunk 3]
```diff
@@ -109,6 +109,14 @@ func (i DockerClient) DeleteImages(images []string, retries int) []error {
 	return errs
 }
 
+func (i DockerClient) RunImage(image string, args ...string) ([]string, error) {
```
Comment (vladimirvivien) [inline pkg/image/docker_client.go:112]: I don't see where this is used in this PR.

**Review by vladimirvivien — COMMENTED** [review]
Let's clarify my comments.

File: kind-config.yaml (hunk 1 above)
Comment (wilsonehusin) [reply kind-config.yaml:5]: I thought it'd be more helpful to make sure that the `control-plane` and `worker` nodes have consistent versions, instead of manually ensuring they are the same.

File: pkg/image/docker_client.go (hunk 3 above)
Comment (wilsonehusin) [reply pkg/image/docker_client.go:112]: `cmd/sonobuoy/app/images.go` under `collectPluginImages`

https://github.com/vmware-tanzu/sonobuoy/pull/1192/files#diff-7adc43f24646df465c7cf42f8867fd929edc34efe0a346af720bc909766e2114R418

File: pkg/image/dryrun_client.go (hunk 2 above)
Comment (wilsonehusin) [reply pkg/image/dryrun_client.go:28]: ah yeah it's just for testing sonobuoy, this is a dryrun client which doesn't actually execute Docker and doesn't test the cluster.

**Review by vladimirvivien — APPROVED** [review]
LGMT
